ui/src/main/java/org/pentaho/di/ui/spoon/tree/extension/TreePaneManager.java

```diff
@@ -0,0 +1,129 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2024 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.ui.spoon.tree.extension;
+
+import org.pentaho.di.core.exception.KettleException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.SashForm;
+import org.eclipse.swt.widgets.Composite;
+
+/** Manages the panes on a tree tab */
+public class TreePaneManager {
+
+  private boolean acceptingExtensions = true;
+  private final List<TreePaneExtension> extensions = new ArrayList<>();
+
+  public void addPane( TreePaneExtension extension ) throws KettleException {
+    if ( acceptingExtensions ) {
```

Comment:
How is order managed? Like, how could a plugin choose whether the new pane is on the top or bottom of the pane?